de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/SessionUtils.java

```diff
@@ -0,0 +1,82 @@
+package de.fu_berlin.inf.dpp.session;
+
+import de.fu_berlin.inf.dpp.SarosPluginContext;
+import de.fu_berlin.inf.dpp.activities.SPath;
+import de.fu_berlin.inf.dpp.filesystem.IResource;
+
+import org.picocontainer.annotations.Inject;
+
+/**
+ * A utility class to provide easy access to functionality needing a session
+ * object.
+ */
+public class SessionUtils {
+    private static volatile ISarosSession currentSarosSession;
+
+    @Inject
+    private static ISarosSessionManager sarosSessionManager;
+
+    static {
+        SarosPluginContext.initComponent(new SessionUtils());
+
+        sarosSessionManager
+            .addSessionLifecycleListener(new NullSessionLifecycleListener() {
+                @Override
+                public void sessionStarted(ISarosSession session) {
+
+                    setSession(session);
+                }
+
+                @Override
+                public void sessionEnded(ISarosSession session,
+                    SessionEndReason reason) {
+
+                    resetSession();
+                }
+            });
+    }
+
+    private static void setSession(ISarosSession newSarosSession) {
+        currentSarosSession = newSarosSession;
+    }
+
+    private static void resetSession() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Move this method into the anonymous class declared at line 23. ([squid:S3398](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS3398))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=293&issues=AWbRiJzSSgN4uCclDZzu&open=AWbRiJzSSgN4uCclDZzu)

[ISSUE_KEY]: AWbRiJzSSgN4uCclDZzu